canlink/tracer.go

```diff
@@ -90,86 +79,31 @@ func WithBusName(name string) TracerOption {
 	}
 }
 
-// WithFiles sets different filetypes the tracer can dump CAN data to
-func WithFiles(f ...TraceFile) TracerOption {
+// WithFrameChannel sets frame channel for incoming frames
+func WithFrameCh(frameCh chan TimestampedFrame) TracerOption {
 	return func(t *Tracer) {
-		for _, filetype := range f {
-			t.types = append(t.types, filetype)
-		}
+		t.frameCh = frameCh
 	}
 }
 
-// Open opens a receiver and spawns a fetchData routine
-func (t *Tracer) Open(ctx context.Context) error {
-	t.l.Info("creating socketcan connection")
-
-	t.receiver = socketcan.NewReceiver(t.conn)
-
-	t.l.Info("canlink receiver created")
-
-	// IMPORTANT: frameCh must be open before isRunning is set
-	t.frameCh = make(chan TimestampedFrame, _frameBufferLength)
-
-	go t.fetchData(ctx)
-
-	return nil
-}
-
-// StartTrace starts receiving and caching CAN frames
-func (t *Tracer) StartTrace(ctx context.Context) error {
-	if !t.isRunning {
-		t.l.Info("received start command")
-
-		t.stop = make(chan struct{})
-
-		go t.receiveData(ctx)
-
-		t.isRunning = true
-
-		t.l.Info("tracer is running")
-	}
-
-	return nil
-}
-
-// StopTrace stops the receiving and caching frames, then dumps them to a file as long as the tracer is not stopped
-func (t *Tracer) StopTrace() error {
-	if t.isRunning {
-		t.l.Info("sending stop signal")
-
-		// IMPORTANT: must set isRunning to false before closing frameCh
-		t.isRunning = false
-
-		close(t.stop)
-
-		t.l.Debug("dumping cached data to files", zap.Int("num_frames", len(t.cachedData)))
-		for _, files := range t.types {
-			err := files.dumpToFile(t.cachedData, t.traceDir, t.busName)
-
-			if err != nil {
-				t.l.Error("failed to dump to file", zap.Error(err))
-
-			}
-		}
-
-		t.cachedData = nil
+// WithConverter sets the converter
+func WithConverter(converter Converter) TracerOption {
```

Comment:
Is a `Converter` required for the tracer? What happens if the user doesn't pass a `WithConverter` option?

Reply:
it should be required, I changed it to a required param